packages/uniforms/src/ValidatedForm.js

```diff
@@ -4,203 +4,194 @@ import merge     from 'lodash/merge';
 import isEqual   from 'lodash/isEqual';
 import set       from 'lodash/set';
 
-import BaseForm from './BaseForm';
+import BaseForm                   from './BaseForm';
+import {__childContextTypesBuild} from './BaseForm';
+import {__childContextTypes}      from './BaseForm';
 
-const Validated = parent => {
+const childContextTypes = __childContextTypesBuild(merge(
+    {state: {validating: PropTypes.bool.isRequired}},
+    __childContextTypes
+));
 
-    const plainChildContextTypes = merge(
-        {
-            uniforms: {
-                state: {
-                    validating: PropTypes.bool
-                }
-            }
-        },
-        parent.plainChildContextTypes
-    );
-
-    return class extends parent {
-        static Validated = Validated;
-
-        static displayName = `Validated${parent.displayName}`;
+const Validated = parent => class extends parent {
+    static Validated = Validated;
 
-        static defaultProps = {
-            ...parent.defaultProps,
+    static displayName = `Validated${parent.displayName}`;
 
-            onValidate (model, error, callback) {
-                callback();
-            },
+    static defaultProps = {
+        ...parent.defaultProps,
 
-            validate: 'onChangeAfterSubmit'
-        };
+        onValidate (model, error, callback) {
+            callback();
+        },
 
-        static propTypes = {
-            ...parent.propTypes,
+        validate: 'onChangeAfterSubmit'
+    };
 
-            onValidate: PropTypes.func.isRequired,
+    static propTypes = {
+        ...parent.propTypes,
 
-            validator: PropTypes.any,
-            validate:  PropTypes.oneOf([
-                'onChange',
-                'onChangeAfterSubmit',
-                'onSubmit'
-            ]).isRequired
-        };
+        onValidate: PropTypes.func.isRequired,
 
+        validator: PropTypes.any,
+        validate:  PropTypes.oneOf([
+            'onChange',
+            'onChangeAfterSubmit',
+            'onSubmit'
+        ]).isRequired
+    };
 
-        static plainChildContextTypes = plainChildContextTypes;
-        static childContextTypes = parent.shapifyPropTypes(plainChildContextTypes);
+    static childContextTypes = {
+        ...parent.childContextTypes || {},
+        uniforms: childContextTypes
```

Comment:
This is awkward because `parent.childContextTypes.uniforms` is replaced with `BaseForm.childContextTypes.uniforms` (through `childContextTypes`) no matter what `parent` is. That is why `BaseForm.plainChildContextTypes` was a part of `BaseForm` (`export __childContextTypesBuild` is still better than `BaseForm.shapifyPropTypes`).

Reply:
I know. There’s a small chance, that it happen not to be the case. Anyway, I’m sure it’s not really important, as the new context API doesn’t have this problem (and I’m thinking about it more and more....).